src/components/forms/Input/index.js

```diff
@@ -209,7 +209,8 @@ module.exports = React.createClass({
       'Input--narrow': width === 'narrow',
       'Input--compact': spacing === 'compact',
       'Input--tight': spacing === 'tight',
-      'Input--loose': spacing === 'loose'
+      'Input--loose': spacing === 'loose',
+      'Input--hasIcon': props.showIcon && (props.icon || state.waiting || state.valid || state.error || props.disabled)
```

Comment:
TBH I think this logic (and the logic in the renderIcon method) should now be abstracted out to a getIcon() method. Then this would read: ```'Input--hasIcon': this.getIcon()